core/src/main/java/org/opensearch/sql/ast/tree/SPath.java

```diff
@@ -0,0 +1,55 @@
+package org.opensearch.sql.ast.tree;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.ast.dsl.AstDSL;
+
+@ToString
+@EqualsAndHashCode(callSuper = false)
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class SPath extends UnresolvedPlan {
+  private UnresolvedPlan child;
+
+  private final String inField;
+
+  @Nullable private final String outField;
+
+  private final String path;
+
+  @Override
+  public UnresolvedPlan attach(UnresolvedPlan child) {
+    this.child = child;
+    return this;
+  }
+
+  @Override
+  public List<UnresolvedPlan> getChild() {
+    return this.child == null ? ImmutableList.of() : ImmutableList.of(this.child);
+  }
+
+  @Override
+  public <T, C> T accept(AbstractNodeVisitor<T, C> nodeVisitor, C context) {
+    return nodeVisitor.visitSpath(this, context);
+  }
+
+  public Eval rewriteAsEval() {
+    String outField = this.outField;
+    if (outField == null) {
+      outField = this.path;
+    }
+
+    return AstDSL.eval(
+        this.child,
+        AstDSL.let(
+            AstDSL.field(outField),
+            AstDSL.function(
+                "json_extract", AstDSL.field(inField), AstDSL.stringLiteral(this.path))));
+  }
```

Comment:
This is the core behavior, the rest is plumbing/parsing boilerplate.

Reply:
It spath is translate to eval, does spath node still required?

Reply:
We're going to be adding more functionality to it later, so I think it's worthwhile to have the stub here. We could probably be more efficient by parsing it directly into an eval at the tree level, though.